app/javascript/components/enrollment/Enrollment.js

```diff
@@ -86,13 +80,31 @@ class Enrollment extends React.Component {
     }
   };
 
-  submit(_event, groupMember, reenableSubmit) {
+  submit = (_event, groupMember, reenableSubmit) => {
     window.onbeforeunload = null;
     axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
 
     let diffKeys = Object.keys(this.state.enrollmentState.patient).filter(
       k => _.get(this.state.enrollmentState.patient, k) !== _.get(this.props.patient, k) || k === 'id'
     );
+
+    // Manually add preferred contact method for new enrollment if it is not changed
+    if (!this.editMode && !diffKeys.includes('preferred_contact_method')) {
+      diffKeys.push('preferred_contact_method');
+    }
+
+    // Manually add close contact keys to diffKeys if enrolling from a close contact
+    // They are passed in as props and thus are not added by triggering the onChange
+    if (!this.editMode && this.props.cc_id) {
+      const closeContactKeys = ['first_name', 'last_name', 'primary_telephone', 'email', 'contact_of_known_case', 'contact_of_known_case_id', 'exposure_notes'];
```

Comment:
I really don't like this because if we add anything to the close_contact model (which we are planning to soon), we'd have to know to update here which is a very easy thing to forget/not know. I think we should be able pre-populate all the fields by just using current.patient as we do already here: https://github.com/SaraAlert/SaraAlert/blob/master/app/javascript/components/enrollment/steps/Contact.js#L34

Reply:
I tried a couple of things with this... we have propagate_fields variable when editing but it's empty when you do an enrollment.  I tried actually passing things in there by creating a `close_contact_subset` but that doesn't actually do anything... even in that case we still had to keep a list of the close contact keys that were prepopulated...

Reply:
Also, the setting of the email thing you mentioned doesn't actually work.  If you enroll a close contact, their email doesn't save which theoretically it should

Reply:
so not totally sure what the best option is here without doing a pretty big overhaul

Reply:
@timwongj any thoughts?

Reply:
> Also, the setting of the email thing you mentioned doesn't actually work.
What do you mean? That's used to set the confirm email field when there's already an email and it does appear to be working. It just doesn't save the actual email field for close contacts but that's because it isn't done there yet.

>so not totally sure what the best option is here without doing a pretty big overhaul
I found a better solution. The whole enrollment vs edit use of this wizard is what is causing issues. [This](https://github.com/SaraAlert/SaraAlert/blob/master/app/javascript/components/enrollment/Enrollment.js#L94) line of code makes sense when editing (only include the ones that are different), but not when enrolling because when enrolling you want to include everything. We should check `this.props.editMode` and if that is false, just include everything in the diff.

I changed it to check that prop and it all works fine. From this:
```
    let diffKeys = Object.keys(this.state.enrollmentState.patient).filter(
      k => _.get(this.state.enrollmentState.patient, k) !== _.get(this.props.patient, k) || k === 'id'
    );
```

to this:
```
  // If enrolling, include ALL fields in diff keys. If editing, only include the ones that have changed.
    let diffKeys = this.props.editMode ? Object.keys(this.state.enrollmentState.patient).filter(
      k => _.get(this.state.enrollmentState.patient, k) !== _.get(this.props.patient, k) || k === 'id'
    ) : Object.keys(this.state.enrollmentState.patient);

```

Reply:
I like this SO MUCH BETTER. just pushed and rebased

Reply:
this will also prevent any further bugs with preset values when doing a new enrollment in the future